src/machine-learning/machine-learning.ts

```diff
@@ -229,68 +232,153 @@ export class MachineLearning implements FirebaseServiceInterface {
   }
 }
 
-
 /**
  * A Firebase ML Model output object.
  */
 export class Model {
-  public readonly modelId: string;
-  public readonly displayName: string;
-  public readonly tags?: string[];
-  public readonly createTime: string;
-  public readonly updateTime: string;
-  public readonly validationError?: string;
-  public readonly published: boolean;
-  public readonly etag: string;
-  public readonly modelHash?: string;
-
-  public readonly tfliteModel?: TFLiteModel;
-
-  constructor(model: ModelResponse) {
-    if (!validator.isNonNullObject(model) ||
-      !validator.isNonEmptyString(model.name) ||
-      !validator.isNonEmptyString(model.createTime) ||
-      !validator.isNonEmptyString(model.updateTime) ||
-      !validator.isNonEmptyString(model.displayName) ||
-      !validator.isNonEmptyString(model.etag)) {
-      throw new FirebaseMachineLearningError(
-        'invalid-server-response',
-        `Invalid Model response: ${JSON.stringify(model)}`);
+  private model: ModelResponse;
+  private readonly client?: MachineLearningApiClient;
+
+  constructor(model: ModelResponse, client: MachineLearningApiClient) {
+    Model.validateModel(model);
+    this.model = model;
+    // Remove '@type' field. We don't need it.
+    if ((this.model as {[key: string]: any})["@type"]) {
```

Comment:
Why is this necessary? Doesn't look like it's exposed anywhere.

Reply:
Discussed. Moved to one location instead of two.